mailbox/elasticsearch/src/test/java/org/apache/james/mailbox/elasticsearch/ElasticSearchIntegrationTest.java

```diff
@@ -218,4 +218,28 @@ public void reasonableLongTermShouldNotBeIgnored() throws Exception {
         assertThat(messageManager.search(new SearchQuery(SearchQuery.bodyContains(reasonableLongTerm)), session))
             .containsExactly(composedMessageId.getUid());
     }
+
+    @Test
+    public void dynamicMappingShouldFailMessageIndexationWithDifferentTypesOnSameCustomField() throws Exception {
+        MailboxPath mailboxPath = MailboxPath.forUser(USERNAME, INBOX);
+        MailboxSession session = MailboxSessionUtil.create(USERNAME);
+        MessageManager messageManager = storeMailboxManager.getMailbox(mailboxPath, session);
+
+        ComposedMessageId customDateHeaderMessageId = messageManager.appendMessage(
+            MessageManager.AppendCommand.builder()
+                .build(ClassLoader.getSystemResourceAsStream("eml/mailCustomDateHeader.eml")),
+            session);
+
+        elasticSearch.awaitForElasticSearch();
+
+        messageManager.appendMessage(
+            MessageManager.AppendCommand.builder()
+                .build(ClassLoader.getSystemResourceAsStream("eml/mailCustomStringHeader.eml")),
+            session);
+
+        elasticSearch.awaitForElasticSearch();
+
+        assertThat(messageManager.search(new SearchQuery(SearchQuery.headerExists("Custom-header")), session))
+            .containsExactly(customDateHeaderMessageId.getUid());
```

Comment:
I bet it is even worst than this: is this mail indexed `at all`?

Please write a second test `indexingShouldNotFailWhenDifferentTypesOnADinamicallyIndexedField` where you then do a listing of all messages present in ElasticSearch...

Reply:
Actually yes I will write something about it, but I just tested on local quickly and... it seems the mail is still indexed, just not on that custom field

Reply:
Please add a test about it, this is important for categorizing the criticity of the aforementioned bug (criticity ranges from CRITICAL!!!! to mwarf,, so nowing on which side of the spectrum we are is important...)